src/state/apis/snapshot/snapshot.ts

```diff
@@ -224,23 +226,25 @@ export const snapshotApi = createApi({
             }, new Set()),
           )
           const delegation = false // don't let people delegate for discounts - ambiguous in spec
-          const votingPowerResults = await Promise.allSettled(
-            evmAddresses.map(async address => {
-              const votingPowerUnvalidated = await getVotingPower(
-                address,
-                '1',
-                strategies,
-                THOR_TIP_014_BLOCK_NUMBER,
-                THORSWAP_SNAPSHOT_SPACE,
-                delegation,
-              )
-              // vp is THOR in crypto balance
-              return bnOrZero(VotingPowerSchema.parse(votingPowerUnvalidated).vp)
-            }),
+
+          const votingPowerResults = await getVotingPower(
+            evmAddresses,
+            '1',
+            strategies,
+            THOR_TIP_014_BLOCK_NUMBER,
+            THORSWAP_SNAPSHOT_SPACE,
+            delegation,
           )
-          const thorHeld = BigNumber.sum(
-            ...votingPowerResults.filter(isFulfilled).map(r => r.value),
-          ).toNumber()
+
+          const scores = ScoresSchema.parse(votingPowerResults).scores
+
+          const thorHeld = scores.reduce((acc: number, scoreByAddress: Record<string, number>) => {
+            const values = Object.values(scoreByAddress)
+
+            if (!values.length) return acc
+
+            return acc + BigNumber.sum(...values.map(bnOrZero)).toNumber()
+          }, 0)
```

Comment:
Non-blocking: We should maintain these scores as BigNumber throughout the reduction here and stringify at the end. `Number.MAX_SAFE_INTEGER` which equals `2^53 - 1` or `9007199254740991`, is only `0.009007199254740991` FOX for example which means it'll be falling back to lossy floating point precision.

<img src="https://github.com/user-attachments/assets/f43d0250-5369-4057-a512-737a90fd2115" width="200">

As this is already an issue, it's non-blocking for this PR - but we should fix it in a follow-up.

Reply:
Kept BigNumber in [0275bec](https://github.com/shapeshift/web/pull/8057/commits/0275becb4da42956c26e33f854a40e300206058a)